Mathlib/CategoryTheory/Enriched/HomCongr.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2024 Nick Ward. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nick Ward
+-/
+import Mathlib.CategoryTheory.Enriched.Ordinary
+
+/-!
+# Congruence of enriched homs
+
+In a `V`-enriched ordinary category `C`, isomorphisms in `C` induce
+isomorphisms between hom-objects in `V`.
+
+The treatment here parallels that for unenriched categories in
+`Mathlib/CategoryTheory/HomCongr.lean` and that for sorts in
+`Mathlib/Logic/Equiv/Defs.lean` (cf. `Equiv.arrowCongr`). Note, however, that
+they construct equivalences between `Type`s and `Sort`s, respectively, while
+in this file we construct isomorphisms between objects in `V`.
+-/
+
+universe v' v u u'
+
+namespace CategoryTheory
+namespace Iso
+
+open Category MonoidalCategory
+
+variable (V : Type u') [Category.{v'} V] [MonoidalCategory V]
+  {C : Type u} [Category.{v} C] [EnrichedOrdinaryCategory V C]
+
+/-- If we have isomorphisms `α : X ≅ X₁` and `β : Y ≅ Y₁` in `C`, then we can
+construct an isomorphism between `V` objects `X ⟶[V] Y` and `X₁ ⟶[V] Y₁`. -/
+@[simp]
+def eHomCongr {X Y X₁ Y₁ : C} (α : X ≅ X₁) (β : Y ≅ Y₁) :
+    (X ⟶[V] Y) ≅ (X₁ ⟶[V] Y₁) where
+  hom := eHomWhiskerRight V α.inv Y ≫ eHomWhiskerLeft V X₁ β.hom
+  inv := eHomWhiskerRight V α.hom Y₁ ≫ eHomWhiskerLeft V X β.inv
+  hom_inv_id := by
+    rw [← eHom_whisker_exchange, assoc]
+    rw [← eHomWhiskerRight_comp_assoc, hom_inv_id, eHomWhiskerRight_id, id_comp]
+    rw [← eHomWhiskerLeft_comp, hom_inv_id, eHomWhiskerLeft_id]
+  inv_hom_id := by
+    rw [← eHom_whisker_exchange, assoc]
+    rw [← eHomWhiskerRight_comp_assoc, inv_hom_id, eHomWhiskerRight_id, id_comp]
+    rw [← eHomWhiskerLeft_comp, inv_hom_id, eHomWhiskerLeft_id]
+
```

Comment:
```suggestion
  hom_inv_id := by
    rw [← eHom_whisker_exchange]
    slice_lhs 2 3 => rw [← eHomWhiskerRight_comp]
    simp [← eHomWhiskerLeft_comp]
  inv_hom_id := by
    rw [← eHom_whisker_exchange]
    slice_lhs 2 3 => rw [← eHomWhiskerRight_comp]
    simp [← eHomWhiskerLeft_comp]

```
There is nothing wrong with your proofs, this is just a suggestion that shows how you can let `simp` handle the trivial stuff like `id_comp`, `assoc`, `hom_inv_id`, etc.